### PR TITLE
Add task template for automatic speech recognition

### PR DESCRIPTION
**Pull request by lewtun**
This PR adds a task template for automatic speech recognition. In this task, the input is a path to an audio file which the model consumes to produce a transcription.

Usage:

```python
from datasets import load_dataset
from datasets.tasks import AutomaticSpeechRecognition

ds = load_dataset("timit_asr", split="train[:10]")
# Dataset({
#     features: ['file', 'text', 'phonetic_detail', 'word_detail', 'dialect_region', 'sentence_type', 'speaker_id', 'id'],
#     num_rows: 10
# })

task = AutomaticSpeechRecognition(audio_file_column="file", transcription_column="text")
ds.prepare_for_task(task)
# Dataset({
#     features: ['audio_file', 'transcription'],
#     num_rows: 10
# })
```

### CODE REVIEW

File: src/datasets/tasks/automatic_speech_recognition.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from typing import ClassVar, Dict
+
+from ..features import Features, Value
+from .base import TaskTemplate
+
+
+@dataclass(frozen=True)
+class AutomaticSpeechRecognition(TaskTemplate):
+    task: str = "automatic-speech-recognition"
+    input_schema: ClassVar[Features] = Features({"audio_file": Value("string")})
```
Comment (lhoestq) [inline src/datasets/tasks/automatic_speech_recognition.py:11]: Note that in the feature we'll have an Audio feature type that will probably have additional parameters (like ClassLabel) such as the sampling rate or the audio format.

Comment (SBrandeis) [inline src/datasets/tasks/automatic_speech_recognition.py:11]: Maybe `audio_file_path` would be more explicit on what the column represent ?

File: tests/test_arrow_dataset.py [hunk 2]
```diff
@@ -2144,6 +2144,39 @@ def test_task_question_answering(self, in_memory):
                 )
                 self.assertDictEqual(features_after_cast, dset.features)
 
+    def test_task_automatic_speech_recognition(self, in_memory):
```
Comment (lhoestq) [inline tests/test_arrow_dataset.py:2147]: don't hesitate to move it outside of the BaseDatasetTest class, as mentioned in #2529

File: src/datasets/tasks/automatic_speech_recognition.py (hunk 1 above)
Comment (SBrandeis) [reply src/datasets/tasks/automatic_speech_recognition.py:11]: Also, paths are not portable between machines a priori. This is probably good enough for now, but at some point, we'll need to replace the `Value("string")` with an `Audio` or `Signal` feature!

File: tests/test_arrow_dataset.py (hunk 2 above)
Comment (lewtun) [reply tests/test_arrow_dataset.py:2147]: yep, will do this here as well!

File: src/datasets/tasks/automatic_speech_recognition.py (hunk 1 above)
Comment (lewtun) [reply src/datasets/tasks/automatic_speech_recognition.py:11]: good to know!

**Review by SBrandeis — COMMENTED** [review]
Thank you @lewtun !

**Comment by lewtun** [comment]
@SBrandeis @lhoestq i've integrated your suggestions, so this is ready for another review :)

**Comment by lhoestq** [comment]
Merging !

**Pull request merged**